BadIdeaFactory/statnews-buildfire
# Add SSO Plugin

This PR adds the first version of the SSO Plugin, including:

- A settings control for app owners to set the Auth0 API credentials (domain and client ID).
- A plugin widget for negotiating with the Auth0 API, determining authentication status, offering the user the ability to login via popup, saving the resulting user object to local storage, and instructing the user to go to the RSS plugin.

It does not yet automatically forward the user to the RSS plugin automatically. Also, the plugin.json file may need further tweaking as we understand its intended contents.

#### Steps to Test

**Prerequisites:** You'll need to have an Auth0 testing app setup (with a test user) and collect its Domain and Client ID. If you don't already have this on hand, get in touch and I'll sort you out.

1. Pull the plugin and load it into a BuildFire SDK `plugins/` directory.
2. Start the SDK with `yarn start`
3. Navigate to the plugin tester (should be [this URL](http://localhost:3000/pluginTester/index.html#/plugin/ssoPlugin))
4. Enter valid Domain and Client ID settings (from Prerequisites) in the left-hand ("Control Settings") section. Save and refresh.
5. Click the Log In button on the right-hand ("Widget") section.
6. Log in to the popup using your test user (from Prerequisites).
7. Note the authentication message on the Widget.
8. Try to track and mis-use all of the above.

#### Related Issues

Affects #3
